templates/theories/linear_algebra.md

```diff
@@ -0,0 +1,106 @@
+and is defined in `linear_algebra.special_linear_group`.
+
+### linear maps and equivalences ###
+
+#### linear_algebra.basic ####
+
+The type `M →[R]ₗ M₂`, or `linear_map R M M₂`, represents `R`-linear maps from the `R`-module `M` to `M₂`.
+These are defined by their action on elements of `M`.
+The type `M ≃[R]ₗ M₂`, or `linear_equiv R M M₂`, is the type of invertible `R`-linear maps from `M` to `M₂`.
+
+The equivalence between matrices and linear maps is formalised in `linear_algebra.matrix`.
+`linear_equiv_matrix'` shows that `matrix.mul_vec` is a linear equivalence between `matrix m n R` and `(n → R) →[R]ₗ (m → R)`.
+The difference between matrices and linear maps is that matrices are in their essence an array of entries
+(which incidentally allows actions such as `matrix.mul_vec`),
+while linear maps are in their essence an action on vectors
+(which incidentally can be represented by a matrix if we have a finite basis).
+If you want to do computations, a matrix is a better choice.
+If you want to do proofs without computations, a linear map is a better choice.
+
+The type `general_linear_group R M` is the group of invertible `R`-linear maps from `M` to itself.
+`general_linear_equiv R M` is the equivalence between `general_linear_group` and `M ≃[R]ₗ M`.
+`special_linear_group.to_GL` is the embedding from the special linear group (of matrices) to the general linear group (of linear maps).
+
+The file `linear_algebra.matrix` connects matrices with linear maps.
+`linear_equiv_matrix'` shows that `matrix.mul_vec` is a linear equivalence between `matrix m n R` and `(n → R) →[R]ₗ (m → R)`.
+In addition, `linear_equiv_matrix` takes a basis `ι` for `M₁` and `κ` for `M₂`
+and gives the equivalence between `R`-linear maps between `M₁` and `M₂` and `matrix ι κ R`.
+If you have an explicit basis for your maps, this equivalence allows you to do calculations such as getting the determinant.
+
+The dual space, consisting of linear maps `M →[R]ₗ R`, is defined in `linear_algebra.dual`.
+
+### bilinear, sesquilinear and quadratic forms ###
+
+#### linear_algebra.bilinear_form ####
+
+For an `R`-module `M`, the type `bilin_form R M` is the type of maps `M → M → R` that are linear in both arguments.
+The equivalence between `bilin_form R M` and maps `M →ₗ[R] M →ₗ[R] R` that are linear in both arguments is called `bilin_linear_map_equiv`.
+A matrix `M` corresponds to a bilinear form that maps vectors `v` and `w` to `row v ⬝ M ⬝ col w`.
+The equivalence between `bilin_form R (n → R)` and `matrix n n R` is called `bilin_form_equiv_matrix`.
+
+#### linear_algebra.sesquilinear_form ####
+
+For an `R`-module `M`, the type `sesq_form R M` is the type of maps `M → M → R` that are linear in the first argument and antilinear in the second.
+Antilinearity for `f : sesq M R` means there is an `I : R → R` such that `f x (a • y) = I a * f x y`, `I (x + y) = I x + I y` and `I (x * y) = I y * I x`.
+
+#### linear_algebra.quadratic_form ####
```

Comment:
I see a malformed section title in https://leanprover-community.github.io/mathlib_docs/linear_algebra/quadratic_form.html

Reply:
Should be fixed in leanprover-community/mathlib#2742.